framework/tlvf/AutoGenerated/src/tlvf/test/tlvVarList.cpp

```diff
@@ -125,7 +128,13 @@ bool tlvTestVarList::add_complex_list(std::shared_ptr<cInner> ptr) {
         TLVF_LOG(ERROR) << "No call to create_complex_list was called before add_complex_list";
         return false;
     }
-    if (ptr->getStartBuffPtr() != (uint8_t *)m_complex_list) {
+    uint8_t *src = (uint8_t *)m_complex_list;
+    if (!m_parse__) {
```

Comment:
I'm not sure if this condition is correct. I think this may cause the error while parsing m_complex_list.

Indeed, while parsing, we also check if the pointer is the one we expected. For the first add, that is the m_complex_list pointer, but for subsequent adds, it should point to the end of the list, not the beginning.